authservice/src/main/java/auth/proximity/authservice/controller/AuthController.java

```diff
@@ -190,7 +198,24 @@ public ResponseEntity<UserInfoResponse> getUserInfo(@RequestParam String email)
         UserInfoResponse userInfoResponse = userService.getUserInfo(email);
         return ResponseEntity.ok(userInfoResponse);
     }
+    @PutMapping("/update/info")
+    ResponseEntity<ResponseDto> updateUserInfo(@AuthenticationPrincipal UserDetailsImpl userDetails, @RequestBody UserUpdateRequest userUpdateRequest) {
+        if (userDetails == null) {
+            return ResponseEntity.status(HttpStatus.UNAUTHORIZED).body(new ResponseDto("401", "Unauthorized: Token is missing or invalid"));
+        }
+        String email = userDetails.getEmail();
+        userService.updateUserInfoByEmail(email, userUpdateRequest);
+        return ResponseEntity.ok(new ResponseDto("200", "User updated successfully"));
+    }
 
-
+    @DeleteMapping("/profile-picture")
+    public ResponseEntity<ResponseDto> deleteProfilePicture(@AuthenticationPrincipal UserDetailsImpl userDetails) {
```

Comment:
This is redundant. Use role base to restrict the endpoint and in that case you wouldn't need to check if the principal is null.

If the principal is null, the authentication/authorization filter will bounce it and it won't even get to ur controller.